src/node/desktop/src/main/detect-r.ts

```diff
@@ -295,7 +295,7 @@ function isValidInstallationWin32(installPath: string): boolean {
 
 function findDefaultInstallPathWin32(version: string): string {
   // query registry for R install path
-  const keyName = `HKEY_LOCAL_MACHINE\\SOFTWARE\\R-Core\\${version}`;
+  const keyName = `HKEY_LOCAL_MACHINE\\SOFTWARE\\R-Core\\R\\${version}`;
```

Comment:
This change looks incorrect to me, given how this function is used:

```
$ rg findDefaultInstallPath
src/node/desktop/src/main/detect-r.ts
296:function findDefaultInstallPathWin32(version: string): string {
329:    const x64InstallPath = findDefaultInstallPathWin32('R64');
338:  const i386InstallPath = findDefaultInstallPathWin32('R');
351:  return findDefaultInstallPathWin32('R');
355:  return findDefaultInstallPathWin32('R64');
```

Note that the caller is responsible for passing `R` or `R64`, since it might depend on whether we're enumerating 32-bit versions of R versus 64-bit versions of R. (The "version" parameter here probably makes this unclear.)

Reply:
Also note that the registry keys `R-core/R` and `R-core/R/3.6.3` both contain InstallPath strings; the first one is just the "default" value associated with the most recently installed copy of R.

Reply:
Interesting... I see some registry keys in both locations as well. I tried re-installing some versions of R, and I'm now always seeing the registry keys updated in the WOW632Node folder, whereas previously some were updated in the parent directory. Did something change in the latest builds of Windows? This seems a little surprising to me.

Reply:
I did a brief comparison of where the registry entries seem to be getting updated:

---

### HKEY_LOCAL_MACHINE\SOFTWARE\R-Core

- R 3.3.3

### HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\R-Core

- R 3.6.3
- R 4.1.3
- R 4.2.0 (Pre-release)

---

Even so, RStudio is still able to find all of these installations...

<img width="406" alt="Screen Shot 2022-03-28 at 10 06 17 AM" src="https://user-images.githubusercontent.com/1976582/160450565-fafba858-96ca-41b3-a06b-ef9c0fce1c02.png">

... however, if I'm running from a command prompt, I need to use `/reg:32` as you said.

```
C:\Users\kevin>reg query HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R /reg:32

HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R
    InstallPath    REG_SZ    C:\R\R-3.3.3
    Current Version    REG_SZ    3.3.3

HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R\3.3.3
HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R\3.6.3
HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R\4.1.3
HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R\4.2.0 Pre-release

C:\Users\kevin>reg query HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R

HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R
    InstallPath    REG_SZ    C:\R\R-3.3.3
    Current Version    REG_SZ    3.3.3

HKEY_LOCAL_MACHINE\SOFTWARE\R-Core\R\3.3.3
```

For posterity, this is with Windows 11 on ARM, running through Parallels on my M1 macOS machine. This seems worth testing on native hardware...

Reply:
Yes, it would be great to test natively. I can only get it to work (Win 11 ARM, Parallels) if the `/reg:32` switch is set and that seems wrong. I also had issues with detecting the architecture. Despite this being an arm64 install, `process.architecture` returns x64.